src/appleseed.studio/mainwindow/renderingsettingswindow.cpp

```diff
@@ -651,14 +666,25 @@ namespace
 
         void create_bounce_settings(QFormLayout* layout, const string& prefix)
         {
-            const string widget_base_key = prefix + ".bounces.";
 
+            const string widget_base_key = prefix + ".bounces.";
             QSpinBox* max_bounces = create_integer_input(widget_base_key + "max_bounces", 0, 10000, 1);
             QCheckBox* unlimited_bounces = create_checkbox(widget_base_key + "unlimited_bounces", "Unlimited");
+            
+            if(prefix == "photon_tracing")
+                max_bounces->setToolTip(m_params_metadata.get_path("sppm.photon_tracing_max_path_length.help"));
+            else
+                max_bounces->setToolTip(m_params_metadata.get_path("pt.max_path_length.help"));
```

Comment:
That also doesn't look right. I think you should construct the path to the `help` component with the prefix, if possible, rather than testing for a few values of the prefix.